source/common/http/status.cc

```diff
@@ -21,6 +21,8 @@ absl::string_view statusCodeToString(StatusCode code) {
     return "PrematureResponseError";
   case StatusCode::CodecClientError:
     return "CodecClientError";
+  default:
```

Comment:
IIRC windows compiler doesn't guarantee all codepaths are hit since `StatusCode` enum has only the above possibilities?

Reply:
maybe a deficiency of MSVC, but this example shows it does not figure it out: https://godbolt.org/z/wAKSeA

Reply:
Please remove the default and add NOT_REACHED as the last line in the function outside of the switch. I think this will compile and it will preserve the error checking around the case.